### ISSUE TITLE
Client side web interface does not fit correctly in HiDPI clients

### ISSUE THREAD

**Issue by ehfd** [selkies-project/selkies-gstreamer#11]
![image](https://user-images.githubusercontent.com/8457324/152354030-3ddb59cd-73af-46ab-80e4-b62fa746b3dc.png)
![image](https://user-images.githubusercontent.com/8457324/152354675-0255eda9-c0b7-4718-b3cd-e7d369bd54df.png)

When not used full-screen, local scaling interface does not fit into the window of the web browser in HiDPI clients. Reproducible by setting PC DPI settings above 100% (96 DPI for Linux). 125% (120 DPI for Linux) requires a slight scroll to see the taskbar, and 150% (144 DPI for Linux) requires more scroll to see the taskbar.

Explore how noVNC and Guacamole implements local scaling.
Starting points:
https://github.com/novnc/noVNC/blob/master/app/ui.js
https://issues.apache.org/jira/browse/GUACAMOLE-164
https://issues.apache.org/jira/browse/GUACAMOLE-165

**Comment by danisla** [comment]
Rather than scaling locally, HiDPI is best supported by changing the remote. Doing local scaling would mean the resulting image would be blurry.

From my initial research, this would have to be done per window manager, I've seen this done before with Mutter.
https://wiki.archlinux.org/title/HiDPI

I am not sure if there is a window manager agnostic way to configure this.

**Comment by ehfd** [comment]
I did more investigation, and this specific issue I'm seeing is with the client HTML5.
![image](https://user-images.githubusercontent.com/8457324/152470791-0f7c130b-172a-4ba0-8d1e-8cb6c9fab2ba.png)
Since the scroll is required since before the screen is started.
Likely will need to look at Javascript.

**Comment by ehfd** [comment]
Should be changed from enhancement to bug...

**Comment by xhejtman** [comment]
Do you use some tool bar like bookmarks bar? I believe this is because of tool bars. For some stupid reason, `document.body.offsetHeight` contains height **including** tool bars which breaks all things. I do not know how to workaround this. It seems that `height: 100%` canvas overflow in this case. It's get fixed once you resize chrome window, at least for me.

**Comment by ehfd** [comment]
Hi! @xhejtman 
Thank you for all your comments to this repository.
I develop https://github.com/ehfd/docker-nvidia-egl-desktop and https://github.com/ehfd/docker-nvidia-glx-desktop and have collaborated with this project.
It would be great if we could receive more helping hands to this repository. The performance of this project as an open source XCloud/Shadow/GeForce Now is already great, and there are more performance and functional improvements that would surely help.
Thanks for your comments again.

**Comment by xhejtman** [comment]
I found fix for this. css/vuetify.css:

set overflow-x to hidden:
```
html {
  box-sizing: border-box;
  overflow-y: hidden; /* All browsers without overlaying scrollbars */
  -webkit-text-size-adjust: 100%; /* iOS 8+ */
}
```

do not use min-height but, height and max-height.
```
.application--wrap {
  flex: 1 1 auto;
  -webkit-backface-visibility: hidden;
          backface-visibility: hidden;
  display: flex;
  flex-direction: column;
  height: 100vh;
  max-height: -webkit-fill-available;
  max-width: 100%;
  position: relative;
}
```

seems to be ok on chrome and safari.

btw, for safari, the following wrap in `if` is needed:
```
if(navigator.permissions) {
    navigator.permissions.query({
        name: 'clipboard-read'
    }).then(permissionStatus => {
        // Will be 'granted', 'denied' or 'prompt':
        if (permissionStatus.state === 'granted') {
            app.clipboardStatus = 'enabled';
        }

        // Listen for changes to the permission state
        permissionStatus.onchange = () => {
            if (permissionStatus.state === 'granted') {
                app.clipboardStatus = 'enabled';
            }
        };
    });
}
```

if safari does not enabled Permission-api, it failes and prevent app to start.

**Comment by ehfd** [comment]
Thank you! 
@xhejtman 
If you make a pull request @danisla can review it and then merge it to main.

**Comment by ehfd** [comment]
Fixed in #33.
